client/apps/edit/components/section_container/index.coffee

```diff
@@ -37,23 +40,70 @@ module.exports = React.createClass
     e.stopPropagation()
     @props.section.destroy()
 
+  onDragStart: (e) ->
+    dragStart = e.clientY - ($(e.currentTarget).position().top - window.scrollY)
+    @props.onDragStart e, dragStart
+
+  onDragOver: (e) ->
+    mousePosition = e.clientY - @props.dragStart
+    $dragOver = $(e.currentTarget).find('.edit-section-container')
+    dragOverID = $dragOver.data('id')
+    dragOverTop = $dragOver.position().top + 20 - window.scrollY
+    dragOverCenter = dragOverTop + ($dragOver.height() / 2)
+    if mousePosition > dragOverCenter and dragOverID is !@props.sections.length or
+     dragOverID is @props.dragging + 1
+      @setState dropPosition: 'bottom'
+    else
+      @setState dropPosition: 'top'
+    @props.onSetDragOver dragOverID unless dragOverID is @props.dragOver
+
+  printDropPlaceholder: ->
+    if @props.dragOver is @props.index
+      unless @props.index is @props.dragging
```

Comment:
Nesting conditional logic like this can lead to hard to follow code. It's often nicer to combine the conditionals with `and`/`or`s. Sometimes it can also help to assign the boolean results to variables for readability like...

```coffeescript
isDragOverSection = @props.dragOver is @props.index
isntDraggedSection = @props.index isnt @props.dragging

if isDragOverSection and isntDraggedSection
  div ...
```

I also noticed some of these variables are numbers, so it might be good to name them to reflect they're indexes, e.g. `dragOverSectionIndex`, `draggingSectionIndex`.

EDIT: I see you did a great job of [this above](https://github.com/artsy/positron/pull/1059/files#diff-4825e94e6420e2a9fc2921e5807397c3R53) 👏